src/Npgsql.NetTopologySuite/NpgsqlNetTopologySuiteExtensions.cs

```diff
@@ -0,0 +1,67 @@
+﻿#region License
+// The PostgreSQL License
+//
+// Copyright (C) 2018 The Npgsql Development Team
+//
+// Permission to use, copy, modify, and distribute this software and its
+// documentation for any purpose, without fee, and without a written
+// agreement is hereby granted, provided that the above copyright notice
+// and this paragraph and the following two paragraphs appear in all copies.
+//
+// IN NO EVENT SHALL THE NPGSQL DEVELOPMENT TEAM BE LIABLE TO ANY PARTY
+// FOR DIRECT, INDIRECT, SPECIAL, INCIDENTAL, OR CONSEQUENTIAL DAMAGES,
+// INCLUDING LOST PROFITS, ARISING OUT OF THE USE OF THIS SOFTWARE AND ITS
+// DOCUMENTATION, EVEN IF THE NPGSQL DEVELOPMENT TEAM HAS BEEN ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//
+// THE NPGSQL DEVELOPMENT TEAM SPECIFICALLY DISCLAIMS ANY WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE. THE SOFTWARE PROVIDED HEREUNDER IS
+// ON AN "AS IS" BASIS, AND THE NPGSQL DEVELOPMENT TEAM HAS NO OBLIGATIONS
+// TO PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR MODIFICATIONS.
+#endregion
+
+using GeoAPI.Geometries;
+using GeoAPI.IO;
+using NetTopologySuite.Geometries;
+using Npgsql.NetTopologySuite;
+using Npgsql.TypeMapping;
+using NpgsqlTypes;
+using System.Data;
+
+namespace Npgsql
+{
+    /// <summary>
+    /// Extension allowing adding the NetTopologySuite plugin to an Npgsql type mapper.
+    /// </summary>
+    public static class NpgsqlNetTopologySuiteExtensions
+    {
+        /// <summary>
+        /// Sets up NetTopologySuite mappings for the PostGIS types.
+        /// </summary>
+        /// <param name="mapper">The type mapper to set up (global or connection-specific)</param>
+        /// <param name="reader">The writer which consumes WKB input</param>
+        /// <param name="writer">The reader which produces WKB output</param>
+        public static INpgsqlTypeMapper UseNetTopologySuite(this INpgsqlTypeMapper mapper, IBinaryGeometryReader reader = null, IBinaryGeometryWriter writer = null)
```

Comment:
Does it really make sense to expose the reader and writer to the user? In what scenario do you expect users to specify these?

Reply:
It makes sense because someone may want to specify parameters for the reader. Same for the reader for symmetry.

```csharp
public PostGisReader(IGeometryFactory factory);
public PostGisReader(ICoordinateSequenceFactory coordinateSequenceFactory, IPrecisionModel precisionModel);
public PostGisReader(ICoordinateSequenceFactory coordinateSequenceFactory, IPrecisionModel precisionModel, Ordinates handleOrdinates);
```

Reply:
OK, makes total sense. And I think it's better exactly as you did it, accepting instances of IBinaryGeometryReader/IBinaryGeometryWriter.

Reply:
Revert or leave the current changes? See the Feedback commit.

Reply:
1. UseNetTopologySuite() should accept IBinaryGeometry{Reader,Writer}, this allows maximum flexibility in case users need to do something odd
2. If none are given, I'd guess the default should be PostGis{Reader,Writer} but I feel like we need to understand the differences between that and the generic WKB{Reader,Writer}. If all your tests pass on PostGis{Reader,Writer} and it seem to work well, let's go with that.

Reply:
Don't know why they need a few implementations, but from my point of view the difference only in bugs (: Will ask.

Reply:
OK, mention me in those issues so I can follow the conversation too. The important thing is for users to be able to specify what they want, and for the default to work well.